tests/vmi_hook_sidecar_test.go

```diff
@@ -218,6 +223,87 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
 			})
 		})
 
+		Context("with sidecar-shim", func() {
+			It("should receive Terminal signal on VMI deletion", func() {
+				vmi = tests.RunVMIAndExpectLaunch(vmi, 360)
+
+				err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Delete(context.Background(), vmi.Name, &metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				Eventually(func() bool {
+					vmiPod, exists, err := getVMIPod(vmi)
+					if err != nil {
+						Expect(err).NotTo(HaveOccurred(), "must be able to retrieve the VMI virt-launcher pod")
+					} else if !exists {
+						return false
+					}
+
+					var tailLines int64 = 100
+					logsRaw, err := virtClient.CoreV1().
+						Pods(vmiPod.GetObjectMeta().GetNamespace()).
+						GetLogs(vmiPod.GetObjectMeta().GetName(), &k8sv1.PodLogOptions{
+							TailLines: &tailLines,
+							Container: sidecarContainerName,
+						}).
+						DoRaw(context.Background())
+					Expect(err).ToNot(HaveOccurred())
+					return strings.Contains(string(logsRaw), "sidecar-shim received signal: terminated")
+				}, 30*time.Second, time.Second).Should(
+					BeTrue(),
```

Comment:
If you expect inside the Eventually func (`g Gomega`), you can match on `Succeeded()` here instead.

Reply:
Thanks, I'm still learning this test framework. I'll change it.